File: ruby/math/prime-seive.rb
input=200
arr = Array.new(10000)
s = Math.sqrt(input)

for i in 2..input
  if arr[i] != 1
    arr[i]=0
  end
  for j in 2..s+2
    if i*j <= input
      arr[i*j]=1
    end
  end
end
    

count=0
for i in 2..input
  if arr[i] == 0
    puts i
    count=count+1
  end
end
    
puts "total: #{count}"

File: js/math/prime-seive.js
var input=200, arr = [];
var s = Math.sqrt(input);

var i,j;
for(i=2; i<=input; i++){
	if(arr[i] != 1){
		arr[i]=0;
	}
	for(j=2; j<s+2; j++){
		if(i*j <= input){
			arr[i*j]=1;
		}
	}
}

var count=0;
for(i=2;i<=input;i++){
	if(arr[i] == 0){
		console.log(i);
		count++;
	}
}
console.log("total: "+count);